device/tt_cluster_descriptor.cpp

```diff
@@ -338,40 +338,6 @@ std::unique_ptr<tt_ClusterDescriptor> tt_ClusterDescriptor::create_from_yaml(con
     return desc;
 }
 
-std::unique_ptr<tt_ClusterDescriptor> tt_ClusterDescriptor::create_for_grayskull_cluster(
```

Comment:
Yeah, unfortunately this is an API change, will need an update in tt_metal. Probably in tt_debuda as well. Please update the main PR comment.

Reply:
You can ignore tt_debuda though, I'll update it (I don't think that they call this function actually). I broke it when I removed CEM, I was mistaken that they don't use it.

Reply:
I can help out with the metal required changes for this

Reply:
Ah actually, there is a flow in tt_metal
```
if (this->target_type_ == TargetDevice::Simulator) {
this->cluster_desc_ =
            tt_ClusterDescriptor::create_for_grayskull_cluster(logical_mmio_device_ids, physical_mmio_device_ids);
}
```

Let's add this to our API tests, so we keep track of it.

Not sure what would be the right solution here, maybe convert this function into a ::create_dummy_cluster(set<chip_id_t>), which would create simplest clusterdescriptor with single chip?

@abhullar-tt what's your take on this?

Re: Changes in tt_metal, if you have the time, sure, but we can also do it (you can just help with the review). If you choose to do it, we're trying to have a process around it where tt_metal PR gets approved before UMD one is merged, so that you can quickly merge the tt_metal counterpart after tt_umd one is merged, therefore we don't block each other. So a PR in tt_metal should be created pointing to this branch, and then switched to UMD main before merging.

Reply:
yeah I think it would be to good to have an api `create_cluster_for_simulation`

Reply:
As discussed offline with @pjanevskiTT , you can leave the create_for_grayskull_cluster for now in the code (this will mean that it will not break build for tt_metal), and in one of the following PRs, we can effectively rename and restructure it to cover only Simulation/Mockup cluster.

Reply:
https://github.com/tenstorrent/tt-umd/issues/286